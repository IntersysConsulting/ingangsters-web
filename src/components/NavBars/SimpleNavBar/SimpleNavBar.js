import React, { Component } from "react";
import Navbar from "react-bootstrap/Navbar";
import NavbarBrand from "react-bootstrap/NavbarBrand";
import { Link } from "react-router-dom";
import "./SimpleNavBar.css";
import "../../../css/colors.css";

class SimpleNavBar extends Component {
  render() {
    return (
      <Navbar
        bg="dark"
        variant="dark"
        sticky="top"
        className="justify-content-center navbar"
      >
        <center>
          <NavbarBrand className="navbar-brand">
            <Link to="/">
              <img
                alt=""
                src="/assets/logo.png"
                width="75"
                height="75"
                className="mx-auto"
              />
            </Link>
          </NavbarBrand>
        </center>
      </Navbar>
    );
  }
}
export default SimpleNavBar;
