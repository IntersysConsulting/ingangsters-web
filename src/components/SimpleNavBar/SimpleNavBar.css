.navbar,
.navbar-default {
  background-color: var(--dark-grey) !important;
  text-align: center;
}

.navbar-brand {
  text-align: center;
  margin: 0, auto;
}
