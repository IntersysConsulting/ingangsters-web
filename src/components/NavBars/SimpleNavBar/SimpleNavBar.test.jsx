import React from "react";
import { shallow, mount } from "enzyme";
import SimpleNavBar from "./SimpleNavBar";
import { Router } from "react-router";
import { createMemoryHistory } from "history";

describe("SimpleNavBar", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<SimpleNavBar />);
    expect(wrapper).toMatchSnapshot();
  });

  it("redirects to home", () => {
    const history = createMemoryHistory();
    const wrapper = mount(
      <Router history={history}>
        <SimpleNavBar />
      </Router>
    );
    wrapper.find("img").simulate("click");
    expect(history.location.pathname).toBe("/");
  });
});
