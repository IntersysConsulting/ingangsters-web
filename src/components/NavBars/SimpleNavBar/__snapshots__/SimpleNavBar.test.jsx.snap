// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleNavBar renders correctly 1`] = `
<Navbar
  bg="dark"
  className="justify-content-center navbar"
  collapseOnSelect={false}
  expand={true}
  sticky="top"
  variant="dark"
>
  <NavbarBrand
    className="navbar-brand"
    href="/"
  >
    <img
      alt=""
      className="mx-auto"
      height="75"
      src="/assets/logo.png"
      width="75"
    />
  </NavbarBrand>
</Navbar>
`;
